src/base/static/js/views/place-form-view.js

```diff
@@ -1,136 +1,259 @@
       'submit form': 'onSubmit',
-      'change input[type="file"]': 'onInputFileChange',
-      'click .category-btn.clickable': 'onCategoryChange',
-      'click .category-menu-hamburger': 'onExpandCategories',
+      'change .shareabouts-file-input': 'onInputFileChange',
+      'change .category-btn': 'onCategoryChange',
+      'click .expansion-icon-container': 'onExpandCategories',
       'click input[data-input-type="binary_toggle"]': 'onBinaryToggle',
-      'click .btn-geolocate': 'onClickGeolocate'
+      'click .btn-geolocate': 'onClickGeolocate',
+      'keyup input[name="url-title"]': 'onUpdateUrlTitle'
     },
-    initialize: function(){
+
+    initialize: function() {
       var self = this;
+
+      Backbone.Events.on("panel:close", this.closePanel, this);
+      this.options.router.on("route", this.resetFormState, this);
+
       this.resetFormState();
       this.placeDetail = this.options.placeConfig.place_detail;
+      this.map = this.options.appView.mapView.map;
+      this.geometryEditorView = this.options.geometryEditorView;
+      this.geometryEnabled = false;
 
       TemplateHelpers.overridePlaceTypeConfig(this.options.placeConfig.items,
         this.options.defaultPlaceTypeName);
       TemplateHelpers.insertInputTypeFlags(this.options.placeConfig.items);
+
+      Gatekeeper.registerCollectionsSet(this.options.collectionsSet);
+
+      this.determineRenderabilityForEachCategory();
     },
-    resetFormState: function() {
-      this.formState = {
-        selectedCategoryConfig: {
-          fields: []
-        },
-        isSingleCategory: false,
-        attachmentData: null,
-        commonFormElements: this.options.placeConfig.common_form_elements || {}
-      }
-    },
-    render: function(isCategorySelected) {
-      var isAdmin = false,
-      self = this,
-      placesToIncludeOnForm = _.filter(this.placeDetail, function(place) { 
+
+    render: function() {
+      this.$el.html(Handlebars.templates['place-form']());
+      this.renderGeometryWarningMessage();
+      
+      var placesToIncludeOnForm = _.filter(this.placeDetail, function(place) { 
         return place.includeOnForm; 
       });
 
-      if (Shareabouts.bootstrapped.currentUser &&
-        _.contains(this.options.placeConfig.administrators, Shareabouts.bootstrapped.currentUser.username)) {
-        isAdmin = true;
-      }
-
-      // if there is only one place to include on form, skip category selection page
       if (placesToIncludeOnForm.length === 1) {
-        this.formState.isSingleCategory = true;
-        isCategorySelected = true;
+
+        // If we only have a single category, skip the category selection phase
         this.formState.selectedCategoryConfig = placesToIncludeOnForm[0];
+        this.renderFormFields();
+      } else {
+        this.renderCategoryButtons();      
       }
-      
-      this.checkAutocomplete();
 
+      return this;
+    },
+
+    renderGeometryWarningMessage: function() {
+      this.$el
```

Comment:
I think the reason I split up all the form pieces into separate templates that get added to the DOM in separate steps is because the form gets re-rendered when the user selects different categories. The different pieces need to be re-rendered in different configurations and in different contexts. There's probably a better way to handles all this though...